app/models/report.rb

```diff
@@ -14,10 +16,35 @@ class Report < ApplicationRecord
 
   enum state: [:inactive, :active]
 
+  def initialize(attributes = nil)
+    super(attributes)
+    self.financial_quarter = current_financial_quarter
+    self.financial_year = current_financial_year
+  end
+
   def activity_must_be_a_fund
     return unless fund.present?
     unless fund.fund?
       errors.add(:fund, I18n.t("activerecord.errors.models.report.attributes.fund.level"))
     end
   end
+
+  private def current_financial_quarter
+    case Date.today.month
+    when 4, 5, 6
+      1
+    when 7, 8, 9
+      2
+    when 10, 11, 12
+      3
+    when 1, 2, 3
+      4
+    end
+  end
+
+  private def current_financial_year
+    today = Date.today
```

Comment:
As you never use other parts of the date, how about just setting `year = Date.today.year` here?

```ruby
year = Date.today.year
year -= 1 if current_financial_quarter == 4
year
```